### ISSUE TITLE
donejs update

### ISSUE THREAD

**Issue by matthewp** [donejs/donejs#535]
It would be nice to have a mechanism to update your project to the next major/minor version of donejs. Consider you are on version `0.15` and you want to upgrade to `0.16`. Currently you'd have to do something like:
### before

``` js
{
  "can-ssr": "1.0.0",
  "done-autorender": "1.0.0"
}
```
### after

``` js
{
  "can-ssr": "2.0.0",
  "done-autorender": "2.0.0"
}
```

You'd have to know which dependencies needed to be updated. In this example can-ssr might have a breaking change that doesn't work with done-autorender 1.0.0, so if you only updated can-ssr your app wouldn't work.

This makes upgrading to minor versions of donejs hard. The way I've done it myself is look at donejs-cli's [package.json](https://github.com/donejs/cli/blob/master/package.json#L53). But users don't know about that (and shouldn't).

Maybe we could have:

``` js
donejs update 0.16
```

Which would upgrade all of your dependencies to 0.16.

**Comment by marshallswain** [comment]
This would be a much easier upgrade path than what I've been doing:
- rename old app folder
- `donejs init app-name`
- copy over old app to new app.

**Comment by matthewp** [comment]
Now that there are codemods in canjs we could have that as part of this.

Probably by default? Can't see why not.

```shell
donejs upgrade
```

Without codemods:

```shell
donejs upgrade --no-codemods
```

**Comment by matthewp** [comment]
Fixed in #1080
